File: ts/shirt.ts
import { IProduct } from "./interfaces";

export default class Shirt implements IProduct {

    private name : string;
    private qty : number;
    private color : string;

    constructor(
        name : string,
        qty : number,
    ) {
        this.name = name;
        this.qty = qty;
    }

    public stocks() : number {
        return this.qty;
    }

   public isAvailable() : boolean {
       return this.qty > 0;
   }
}

File: shirt.go
package main

// Shirt ...
type Shirt struct {
	Name string `json:"name,omitempty"`
	Qty  int    `json:"qty,omitempty"`
	Size string
}

// Stocks check reminding stocks
func (b *Shirt) Stocks() int {
	return b.Qty
}

// IsAvailable check reminding stocks
func (b *Shirt) IsAvailable() bool {
	return b.Qty > 0
}

// NewShirt ...
func NewShirt(b Shirt) Product {
	return &b
}
